### PR TITLE
In payments status, show next contribution date

### LINKED ISSUE #3801: Add notification under status message to inform user of their contribution day
Our Brave Payments users need to know when their first, and next payment will post. This info can be added below the Status message as shown below.

image

This "next payment" status will be shown as soon as the wallet detects a balance for the first time.

The date will change based on the users usage based on the start date of first wallet funding or from the last payment date.

### PR DESCRIPTION
**Pull request by ayumi**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Fix #3801

Auditors: @bradleyrichter @diracdeltas

Test Plan:
1. Open Brave to Preferences > Payments
2. Confirm the "status" column has "Next contribution: {date}" on the bottom

First run
screen shot 2016-09-08 at 16 55 43
Created a wallet
screen shot 2016-09-08 at 16 55 54
Funded 💰
screen shot 2016-09-08 at 16 55 06

### CODE REVIEW

File: js/about/preferences.js [hunk 1]
```diff
@@ -686,6 +686,21 @@ class PaymentsTab extends ImmutableComponent {
     </div>
   }
 
+  get nextReconcileDate () {
```
Comment (diracdeltas) [inline js/about/preferences.js]: this should return null if `ledgerData.get('reconcileStamp')` is not a number (may be undefined/null)

File: js/about/preferences.js [hunk 2]
```diff
@@ -686,6 +686,21 @@ class PaymentsTab extends ImmutableComponent {
...
+    let ledgerData = this.props.ledgerData
+    let balance = Number(ledgerData.get('balance') || 0)
+    if (balance === 0 || !ledgerData.get('reconcileStamp')) {
+      return null
+    }
+    let nextReconcileDate = formattedDateFromTimestamp(ledgerData.get('reconcileStamp'))
+    let l10nDataArgs = {
+      reconcileDate: nextReconcileDate
+    }
+    return <div className='nextReconcileDate' data-l10n-args={JSON.stringify(l10nDataArgs)} data-l10n-id='statusNextReconcileDate' />
```
Comment (diracdeltas) [inline js/about/preferences.js]: we generally prefer `const` over `let` for declaring variables that aren't going to be reassigned

File: js/about/preferences.js (hunk 1 above)
Comment (ayumi) [inline js/about/preferences.js]: i think I check for this on line 694 – is that right?

Comment (diracdeltas) [inline js/about/preferences.js]: oops i missed that, sorry

File: js/about/preferences.js (hunk 2 above)
Comment (ayumi) [inline js/about/preferences.js]: thx good idea – fixed it

File: js/about/preferences.js [hunk 3]
```diff
@@ -686,6 +686,21 @@ class PaymentsTab extends ImmutableComponent {
     </div>
   }
 
+  get nextReconcileDate () {
+    // Show date only after the user has funded the wallet.
+    // Afterwards balance should always !== 0 due to exchange rate fluctuations.
+    const ledgerData = this.props.ledgerData
+    const balance = Number(ledgerData.get('balance') || 0)
+    if (balance === 0 || !ledgerData.get('reconcileStamp')) {
```
Comment (diracdeltas) [inline js/about/preferences.js]: actually i think this should show up even if the balance is 0. there is sometimes a long delay between when the wallet is funded and when it actually shows up as balance.

Comment (ayumi) [inline js/about/preferences.js]: i feel it – updated

**Comment by diracdeltas** [comment]
sweet thanks

**Pull request merged**